<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="huffmanTab">
       <attribute name="title">
        <string>Хаффман</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1">
        <property name="spacing">
         <number>11</number>
        </property>
        <item row="0" column="1">
         <widget class="QGroupBox" name="encodeBox">
          <property name="title">
           <string>Кодирование</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,0,1,0,0">
           <property name="leftMargin">
            <number>11</number>
           </property>
           <property name="topMargin">
            <number>11</number>
           </property>
           <property name="rightMargin">
            <number>11</number>
           </property>
           <property name="bottomMargin">
            <number>11</number>
           </property>
           <item>
            <widget class="QTextEdit" name="originalTextHuffman">
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;there are kinds of symbols available, then so&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="placeholderText">
              <string>Исходный текст</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="encodeButton">
             <property name="text">
              <string>Закодировать</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextEdit" name="encodedText">
             <property name="placeholderText">
              <string>Закодированный текст</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="entropy">
             <property name="text">
              <string>Энтропия:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="avgCodeLength">
             <property name="text">
              <string>Средняя длина кода:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0" rowspan="3">
         <widget class="QGroupBox" name="codeTable">
          <property name="title">
           <string>Таблица кодирования</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <property name="leftMargin">
            <number>11</number>
           </property>
           <property name="topMargin">
            <number>11</number>
           </property>
           <property name="rightMargin">
            <number>11</number>
           </property>
           <property name="bottomMargin">
            <number>11</number>
           </property>
           <item>
            <widget class="QTableWidget" name="huffmanTable">
             <property name="font">
              <font>
               <pointsize>9</pointsize>
              </font>
             </property>
             <property name="sortingEnabled">
              <bool>false</bool>
             </property>
             <property name="cornerButtonEnabled">
              <bool>false</bool>
             </property>
             <property name="columnCount">
              <number>3</number>
             </property>
             <column>
              <property name="text">
               <string>Символ</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Количество</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Код</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QGroupBox" name="decodeBox">
          <property name="title">
           <string>Декодирование</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QTextEdit" name="encodedTextPre">
             <property name="placeholderText">
              <string>Закодированный текст</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="decodeButton">
             <property name="text">
              <string>Раскодировать</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextEdit" name="decodedText">
             <property name="placeholderText">
              <string>Раскодированный текст</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton">
          <property name="font">
           <font>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>Переместить закодированный текст вниз</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="crcTab">
       <attribute name="title">
        <string>CRC</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0,1">
        <property name="spacing">
         <number>11</number>
        </property>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Вычисление</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Исходные данные</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="originalTextCRC">
             <property name="plainText">
              <string>6B 69 6E 64</string>
             </property>
             <property name="placeholderText">
              <string>Введите исходные данные</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Данные в двоичном виде</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="originalTextCRCBin">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="undoRedoEnabled">
              <bool>false</bool>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="backgroundVisible">
              <bool>false</bool>
             </property>
             <property name="placeholderText">
              <string>Здесь появятся исходные данные в двоичном виде при вычислении CRC</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Полином</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="polynomProcess">
             <property name="text">
              <string>x^8+x^7+x^6+x^4+x^2+1</string>
             </property>
             <property name="placeholderText">
              <string>Введите полином в виде x^8+x^6+x^2+1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Полином в двоичном виде</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="polynomProcessBin">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="frame">
              <bool>true</bool>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Здесь появится полином в двоичном виде после ввода</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="calculateCRCButton">
             <property name="text">
              <string>Вычислить CRC</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,1,1,2">
             <item>
              <widget class="QLabel" name="label_12">
               <property name="text">
                <string>CRC (hex)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="resultCRCHex"/>
             </item>
             <item>
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>CRC(bin)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="resultCRCBin"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="moveRight">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>40</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>-&gt;</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Проверка</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="2" column="0" colspan="2">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Полином</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Остаток:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Данные</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="2">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Полином в двоичном виде</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPlainTextEdit" name="checkText">
             <property name="placeholderText">
              <string>Введите данные в двоичном виде</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QLineEdit" name="polynomCheck">
             <property name="text">
              <string>x^8+x^7+x^6+x^4+x^2+1</string>
             </property>
             <property name="placeholderText">
              <string>Введите полином в виде x^8+x^6+x^2+1</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0" colspan="2">
            <widget class="QLineEdit" name="polynomCheckBin">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>Здесь появится полином в двоичном виде после ввода</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLineEdit" name="crcCheckResult"/>
           </item>
           <item row="6" column="0" colspan="2">
            <widget class="QPushButton" name="checkCRCButton">
             <property name="text">
              <string>Проверить CRC</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
